src/lib/user-performance-metrics.ts

```diff
@@ -0,0 +1,138 @@
+// Forked and modified from UXM: https://github.com/treosh/uxm
+
+declare let PerformancePaintTiming: any
+
+const perf = typeof window !== "undefined" ? window.performance : null
+
+const sanitizedMetrics = (start, end) =>
+  start > 0 && end > 0 && end - start > 0 ? end - start : null
+
+// user timing helpers
+
+export function mark(markName) {
+  if (perf && perf.mark) {
+    perf.mark(markName)
+  }
+}
+
+export function measure(measureName, startMarkName) {
+  if (perf && perf.measure) {
+    try {
+      perf.measure(measureName, startMarkName)
+    } catch (err) {
```

Comment:
Is there any specific exceptions that `measure` may throw? If so, can you maybe add a comment that describes them or a link to a definitive doc?

Reply:
Yeah, I can add some comments here. If the markNames specified don't exist (i.e. the mark hasn't been created) then it'll throw an exception.

Reply:
Better yet, maybe add a test that describes that situation?

Reply:
The test would be a little harder because it would need to run in a browser environment to actually be reliable. I'll add a comment though.